modules/iam-assumable-role/main.tf

```
@@ -79,7 +79,7 @@ resource "aws_iam_role" "this" {
   force_detach_policies = var.force_detach_policies
   permissions_boundary  = var.role_permissions_boundary_arn
 
-  assume_role_policy = var.role_requires_mfa ? data.aws_iam_policy_document.assume_role_with_mfa.json : data.aws_iam_policy_document.assume_role.json
+  assume_role_policy = var.custom_role_trust_policy != "" ? var.custom_role_trust_policy : var.role_requires_mfa ? data.aws_iam_policy_document.assume_role_with_mfa.json : data.aws_iam_policy_document.assume_role.json
```

Something like this is more in spirit of Terraform 0.13+:
```suggestion
  assume_role_policy = coalesce(var.custom_role_trust_policy, try(data.aws_iam_policy_document.assume_role_with_mfa[0].json, data.aws_iam_policy_document.assume_role[0].json))
```

---

Also, add `count = var.role_requires_mfa ? 1 : 0` (and opposite) in both data sources.

---

Thanks for the quick review, will try to fix it all today

---

Done both and fixed doc failed check